skimage/feature/util.py

```diff
@@ -145,22 +147,30 @@ def plot_matched_features(
     ax.imshow(image, cmap='gray')
     ax.axis((0, image0.shape[1] + offset[1], image0.shape[0] + offset[0], 0))
 
-    rng = np.random.default_rng()
+    number_of_matches = matches.shape[0]
 
-    for i in range(matches.shape[0]):
-        idx0 = matches[i, 0]
-        idx1 = matches[i, 1]
+    from matplotlib.colors import is_color_like
 
-        if matches_color is None:
-            color = rng.random(3)
-        else:
-            color = matches_color
+    if matches_color is None:
+        rng = np.random.default_rng(seed=0)
+        colors = [rng.random(3) for _ in range(number_of_matches)]
+    elif is_color_like(matches_color):
+        colors = [matches_color for _ in range(number_of_matches)]
+    elif hasattr(matches_color, "__len__") and len(matches_color) > 0:
+        # No need to check each color, matplotlib does so for us
+        color_iter = cycle(matches_color)
+        colors = [next(color_iter) for _ in range(number_of_matches)]
```

Comment:
Why not just always follow this code path, and avoid the checks?

Reply:
Do you mean that we should remove the possibility to pass a single color and only accept a sequence (which can contain only a single element)?

Reply:
My understanding of @stefanv's comment is that we should turn the single color into a sequence of length 1:  
```py
if not hasattr(matches_color, '__iter__'):
    matches_color = [matches_color]
```
and then, indeed, it's enough to:
```py
color_iter = cycle(matches_color)
colors = [next(color_iter) for _ in range(number_of_matches)]
```
assuming we've previously handled the default case where `matches_color is None`.

Reply:
@mkcor Ok understood. I changed the code to use cycle for single color and sequences.

I also adapted the code so that an empty sequence will fall back to random color selection. Otherwise `cycle` would cause a `StopIteration`. If this fall behaviour is too confusing, I can also raise a custom `ValueError`.

Reply:
If we keep the cycling support, a sequence that's empty should definitely raise a ValueError, and a sequence of length one should probably raise a warning. 

I don't feel like we get a clearer API here by supporting cycling. :sweat_smile:

Reply:
As stated in https://github.com/scikit-image/scikit-image/pull/7541#discussion_r1773838427, the idea to cycle the colors has been dropped.